contrail-ui/src/utils/auth-utils.ts

```diff
@@ -18,61 +21,75 @@ export function initializeFirebase() {
         appId: "1:342081308461:web:229362ba93e13630",
     };
     firebase.initializeApp(firebaseConfig);
-    console.log("firebase initialized");
 }
 
 /**
  * Registers a user to Firebase. *NOTE*: {@link initializeFirebase} must be called before.
  * @param displayName the user's display name received from registration
  * @param email the user's email received from registration.
  * @param password the user's password received from registration.
- *
+ * @returns a promise that resolves with the current user, or rejects with the error
  */
 export function registerUser(displayName: string, email: string, password: string) {
     console.log("registering user");
-    firebase.auth().createUserWithEmailAndPassword(email, password).then(function() {
-        console.log("registration successful, updating profile");
-        const user = firebase.auth().currentUser;
-        if(user === null){
-            throw Error('User not found');
-        }
-        user.updateProfile({
-            displayName,
-        }).then(function() {
-            console.log("profile update successful");
+    return new Promise((resolve, reject) => {
+        firebase.auth().createUserWithEmailAndPassword(email, password).then(() => {
+            const user = firebase.auth().currentUser;
+            if (user === null) {
+                console.log("registration failed: user is null");
+                reject("user is null");
+            } else {
+                console.log("registration successful: updating profile");
+                user.updateProfile({
+                    displayName,
+                }).then(() => {
+                    console.log("profile updat successful");
+                    resolve(user);
+                }).catch((error) => {
+                    console.log("profile update failed:  ", error);
+                    reject(error);
+                });
+            }
         });
-
-    }).catch(function(error) {
-        console.log("registration failed: ", error);
     });
 }
 
 /**
  * Logins a user to Firebase. *NOTE* {@link initializeFirebase} must be called before.
  * @param email the user's email received from login
  * @param password the user's password received from login
+ * @return a promise that resolves with the current user, or rejects with the error
  */
 export function loginUser(email: string, password: string) {
```

Comment:
export function loginUser(email: string, password: string) : Promise<${type of user}> {

Reply:
done